### PR TITLE
disable certificate validation via ENV variable

### PR DESCRIPTION
**Pull request by johakoch**
Set `DISABLE_CERTIFICATE_VALIDATION = true` to disable certificate validation

### CODE REVIEW

**Review by filex — CHANGES_REQUESTED** [review]
We should change the variable names. not only the new one, but all variables that affect the couper `backend` definition:

```
ORIGIN -> BACKEND_ORIGIN
ORIGIN_HOSTNAME -> BACKEND_HOSTNAME
*_TIMEOUT -> BACKEND_*_TIMEOUT
DISABLE_CERTIFICATE_VALIDATION -> BACKEND_DISABLE_CERTIFICATE_VALIDATION
```

File: couper.hcl [hunk 1]
```diff
@@ -10,11 +10,12 @@ server "oidc-gate" {
...
-        hostname = env.ORIGIN_HOSTNAME != "" ? env.ORIGIN_HOSTNAME : request.host
-        connect_timeout = env.CONNECT_TIMEOUT
-        ttfb_timeout = env.TTFB_TIMEOUT
-        timeout = env.TIMEOUT
+        origin = env.BACKEND_ORIGIN
+        hostname = env.BACKEND_HOSTNAME != "" ? env.BACKEND_HOSTNAME : request.host
+        connect_timeout = env.BACKEND_CONNECT_TIMEOUT
+        ttfb_timeout = env.BACKEND_TTFB_TIMEOUT
+        timeout = env.BACKEND_TIMEOUT
+        disable_certificate_validation = env.BACKEND_DISABLE_CERTIFICATE_VALIDATION
```
Comment (malud) [inline couper.hcl:18]: The environment string mapping from string `true` or `false` to a boolean one is supported for the `settings` block only. So in this case the assignment must be slightly changed.
```suggestion
        disable_certificate_validation = env.BACKEND_DISABLE_CERTIFICATE_VALIDATION == "true"
```

File: README.md [hunk 2]
```diff
@@ -23,11 +23,12 @@ You need to register a **confidential** client at the OpenID Provider with the r
...
-| `CONNECT_TIMEOUT` | duration | `10s` | The total timeout for dialing and connect to the origin | - |
-| `TTFB_TIMEOUT` | duration | `60s` | The duration from writing the full request to the origin and receiving the answer | - |
-| `TIMEOUT` | duration | `300s` | The total deadline duration a backend request has for write and read/pipe | - |
+| `BACKEND_ORIGIN` | string | - | The origin of the service to be protected | `https://www.example.com` |
+| `BACKEND_HOSTNAME` | string | - | The value of the HTTP host header field for the request to the protected service | - |
+| `BACKEND_CONNECT_TIMEOUT` | duration | `10s` | The total timeout for dialing and connect to the origin | - |
+| `BACKEND_TTFB_TIMEOUT` | duration | `60s` | The duration from writing the full request to the origin and receiving the answer | - |
+| `BACKEND_TIMEOUT` | duration | `300s` | The total deadline duration a backend request has for write and read/pipe | - |
+| `BACKEND_DISABLE_CERTIFICATE_VALIDATION` | bool | `false` | Disables the peer certificate validation for the protected service | - |
```
Comment (malud) [inline README.md:31]:
```suggestion
| `BACKEND_DISABLE_CERTIFICATE_VALIDATION` | string | `false` | Disables the peer certificate validation for the protected service | - |
```

File: couper.hcl [hunk 3]
```diff
@@ -102,10 +103,11 @@ defaults {
...
-    ORIGIN_HOSTNAME = ""
-    CONNECT_TIMEOUT = "10s"
-    TTFB_TIMEOUT = "60s"
-    TIMEOUT = "300s"
+    BACKEND_ORIGIN = ""
+    BACKEND_HOSTNAME = ""
+    BACKEND_CONNECT_TIMEOUT = "10s"
+    BACKEND_TTFB_TIMEOUT = "60s"
+    BACKEND_TIMEOUT = "300s"
+    BACKEND_DISABLE_CERTIFICATE_VALIDATION = false
```
Comment (malud) [inline couper.hcl:111]:
```suggestion
    BACKEND_DISABLE_CERTIFICATE_VALIDATION = "false"
```

**Review by malud — CHANGES_REQUESTED** [review]

**Pull request merged**